fiat-roles/src/main/java/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```diff
@@ -460,20 +460,14 @@ public void remove(@NonNull String id) {
   }
 
   private Set<String> scanSet(byte[] key) {
-    final Set<String> results = new HashSet<>();
-    final AtomicReference<byte[]> cursor =
-        new AtomicReference<>(ScanParams.SCAN_POINTER_START_BINARY);
-    do {
-      final ScanResult<byte[]> result =
-          redisClientDelegate.withBinaryClient(
-              jedis -> {
-                return jedis.sscan(key, cursor.get());
-              });
-      results.addAll(
-          result.getResult().stream().map(SafeEncoder::encode).collect(Collectors.toList()));
-      cursor.set(result.getCursorAsBytes());
-    } while (!Arrays.equals(cursor.get(), ScanParams.SCAN_POINTER_START_BINARY));
-    return results;
+    return redisClientDelegate
+        .withBinaryClient(
+            jedis -> {
+              return jedis.smembers(key);
```

Comment:
Isn't this going to cause Redis to block for the duration of the `SMEMBERS` call?

Reply:
Yes, almost all redis commands block IIRC, and this one is definitely included. However, even with tens of thousands of set members, this happens in <1msec in our case (and from prior experience this should be typical of redis). Compare that to the extreme overhead of having individual commands run for each set member, and it's a clear win overall.

This change also improves the consistency as the set can change from under redis between calls, e.g. this from the scan docs:

> * A given element may be returned multiple times. It is up to the application to handle the case of duplicated elements, for example only using the returned elements in order to perform operations that are safe when re-applied multiple times.
> * Elements that were not constantly present in the collection during a full iteration, may be returned or not: it is undefined.

Finally, given fiat itself locks certain operations while doing a roles sync, I don't think there is too much risk here. Fiat won't be responding to certain requests (and therefore, running other redis commands) anyway until the sync process is finished.